polygon/heimdall/event_record.go

```diff
@@ -48,6 +52,25 @@ func (e *EventRecordWithTime) BuildEventRecord() *EventRecord {
 	}
 }
 
+func UnpackEventRecordWithTime(stateContract abi.ABI, encodedEvent rlp.RawValue) *EventRecordWithTime {
+	commitStateInputs := stateContract.Methods["commitState"].Inputs
+	methodId := stateContract.Methods["commitState"].ID
+
+	if bytes.Equal(methodId, encodedEvent[0:4]) {
+		t := time.Unix((&big.Int{}).SetBytes(encodedEvent[4:36]).Int64(), 0)
+		args, _ := commitStateInputs.Unpack(encodedEvent[4:])
+
+		if len(args) == 2 {
+			var eventRecord EventRecord
+			if err := rlp.DecodeBytes(args[1].([]byte), &eventRecord); err == nil {
```

Comment:
please avoid swallowing errors like this - if there is an error we should return it up the stack